controllers/sr_controller/sr/robot/ruggeduino.py

```diff
@@ -17,36 +17,63 @@ def init_ruggeduino_array(webot):
         "left finger sensor",
         "right finger sensor",
     ]
+    led_names = [
+        "led 1",
+        "led 2",
+        "led 3",
+        "led 4",
+        "led 5",
+        "led 6",
+    ]
+
+    analogue_input_array = [DistanceSensor(webot, name) for name in dist_sensor_names]
 
-    analogue_array = [DistanceSensor(webot, name) for name in dist_sensor_names]
+    digital_input_array = [Microswitch(webot, name) for name in switch_names]
 
-    digital_array = [Microswitch(webot, name) for name in switch_names]
+    digital_output_array = [Led(webot, name) for name in led_names]
 
-    return [Ruggeduino(webot, analogue_array, digital_array)]
+    return [Ruggeduino(webot, analogue_input_array, digital_input_array, digital_output_array)]
 
 
 class Ruggeduino(object):
 
-    DIGITAL_PIN_OFFSET = 2  # Exclude pins 0 and 1 as they are used for USB serial comms
+    DIGITAL_READ_PIN_START = 2  # Exclude pins 0 and 1 as they are used for USB serial comms
+    DIGITAL_WRITE_PIN_START = 7   # Offset output pins by input pins
+    DIGITAL_END_PIN = 12
```

Comment:
This is a reasonable approach, though it might be somewhat more pythonic (and require less effort keeping things in step) to just let invalid requests fail.

It's still possible to provide a useful error message if we want to do that:
``` python
try:
    element = self._list[index - self.OFFSET]
except KeyError:
    raise ValueError("{!r} is not a valid pin (only pins {} - {})".format(
        index,
        self.OFFSET,
        self.OFFSET + len(self._list),
    )) from None

element.whatever()
```
This avoids needing to know exactly what the length of the list is (and thus removes the possibility that it could get out of step) while still providing a clear error message.